packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -283,88 +286,103 @@ const DomainPicker: FunctionComponent< Props > = ( {
-									<SuggestionItem
-										key={ existingSubdomain?.domain_name }
-										domain={ existingSubdomain?.domain_name }
-										cost="Free"
-										isFree
-										isExistingSubdomain
-										railcarId={ baseRailcarId ? `${ baseRailcarId }${ 0 }` : undefined }
-										onRender={ () =>
-											handleItemRender(
-												existingSubdomain?.domain_name,
-												`${ baseRailcarId }${ 0 }`,
-												0,
-												false
-											)
-										}
-										selected={ currentDomain?.domain_name === existingSubdomain?.domain_name }
-										onSelect={ () => {
-											onExistingSubdomainSelect?.( existingSubdomain?.domain_name );
-										} }
-										type={ itemType }
-									/>
-								) }
-							</ItemGrouper>
-							{ segregateFreeAndPaid && (
-								<p className="domain-picker__suggestion-group-label">
-									{ __( 'Professional domains', __i18n_text_domain__ ) }
-								</p>
-							) }
-							<ItemGrouper groupItems={ segregateFreeAndPaid }>
-								{ ( ! areDependenciesLoading &&
-									domainSuggestions?.map( ( suggestion, i ) => {
-										const index = existingSubdomain?.domain_name ? i + 1 : i;
-										const isRecommended = index === 1;
-										const availabilityStatus =
-											domainAvailabilities[ suggestion?.domain_name ]?.status;
-										// should availabilityStatus be falsy then we assume it is available as we have not checked yet.
-										const isAvailable = availabilityStatus
-											? domainIsAvailableStatus?.indexOf( availabilityStatus ) > -1
-											: true;
-										return (
-											<SuggestionItem
-												key={ suggestion.domain_name }
-												isUnavailable={ ! isAvailable }
-												domain={ suggestion.domain_name }
-												cost={ suggestion.cost }
-												isLoading={
-													currentDomain?.domain_name === suggestion.domain_name &&
-													isCheckingDomainAvailability
-												}
-												hstsRequired={ suggestion.hsts_required }
-												isFree={ suggestion.is_free }
-												isRecommended={ isRecommended }
-												railcarId={ baseRailcarId ? `${ baseRailcarId }${ index }` : undefined }
-												onRender={ () =>
-													handleItemRender(
-														suggestion.domain_name,
-														`${ baseRailcarId }${ index }`,
-														index,
-														isRecommended
-													)
-												}
-												onSelect={ () => {
-													onDomainSelect( suggestion );
-												} }
-												selected={ currentDomain?.domain_name === suggestion.domain_name }
-												type={ itemType }
-											/>
-										);
-									} ) ) ||
-									times( neededPlaceholdersCount, ( i ) => (
-										<SuggestionItemPlaceholder type={ itemType } key={ i } />
-									) ) }
-							</ItemGrouper>
-						</>
+						{ groupOrder.map( ( group: DomainGroups ) => {
+							if ( group === 'sub-domain' && existingSubdomain ) {
```

Comment:
By using directly the static `existingSubdomain` instead of grouping it with the suggestions, we can render it independently and not replace it with a placeholder for each new domain search.
Demo: https://cloudup.com/cUxaRzyXiyI

cc: @StefanNieuwenhuis @tjcafferkey